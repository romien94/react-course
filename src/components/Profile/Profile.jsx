import React from "react";
import Header from "../Header/Header";
import PaymentForm from "../PaymentForm/PaymentForm";

class Profile extends React.Component {
  render() {
    return (
      <div>
        <Header />
        <div className="app-profile">
          <div className="app-profile__top">
            <h2 className="app-profile__title">Профиль</h2>
            <p className="app-profile__text">Способ оплаты</p>
          </div>
          <PaymentForm />
        </div>
      </div>
    );
  }
}

export default Profile;
