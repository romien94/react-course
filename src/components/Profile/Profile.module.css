.profile {
    display: flex;
    flex-direction: column;
    width: 945px;
    min-height: 565px;
    background-color: white;
    padding: 75px;
    align-items: center;
    text-align: center;
}

.profileTop {
    margin-bottom: 35px;
}

.profileTitle {
    font-size: 36px;
}

.profileText {
    font-size: 16px;
    color: #505050;
    text-align: center;
}