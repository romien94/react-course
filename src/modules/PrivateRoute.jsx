import React from "react";
import { Route, Redirect } from "react-router-dom";
import { connect } from "react-redux";
import PropTypes from "prop-types";

export const PrivateRoute = connect((state) => ({
  isLoggedIn: state.auth.isLoggedIn,
}))(({ component: Component, isLoggedIn, ...rest }) => {
  propTypes = {
    isLoggedIn: PropTypes.bool,
  };

  return (
    <Route
      {...rest}
      render={(props) =>
        isLoggedIn ? <Component {...props} /> : <Redirect to="/login" />
      }
    />
  );
});
