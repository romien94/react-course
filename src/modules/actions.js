export const LOG_IN = "LOG_IN";
export const LOG_OUT = "LOG_OUT";
export const AUTHENTICATE = "AUTHENTICATE";
export const LOAD_PROFILE = "LOAD_PROFILE";
export const SAVE_PROFILE = "SAVE_PROFILE";
export const SAVE_TO_STORE = "SAVE_TO_STORE";
export const ERROR = "ERROR";
export const REGISTRATION = "REGISTRATION";
export const FETCH_ADDRESSES = "FETCH_ADDRESSES";
export const GET_ADDRESSES = "GET_ADDRESSES";
export const SEND_ROUTE = "SEND_ROUTE";
export const SAVE_COORDS = "SAVE_COORDS";

export const logIn = (token) => ({ type: LOG_IN, payload: token });
export const logOut = () => ({ type: LOG_OUT });

export const authenticate = (email, password) => ({
  type: AUTHENTICATE,
  payload: { email, password },
});

export const loadProfile = (token) => ({
  type: LOAD_PROFILE,
  payload: token
}) 

export const saveProfile = (number, date, name, cvc, token) => ({
  type: SAVE_PROFILE,
  payload: { number, date, name, cvc, token },
});

export const saveCardToStore = (number, date, name, cvc, token) => ({
  type: SAVE_TO_STORE,
  payload: { number, date, name, cvc, token },
});

export const showError = (error) => ({
  type: ERROR,
  payload: error
});

export const registerUser = (email, password, firstName, secondName) => ({
  type: REGISTRATION,
  payload: {email, password, firstName, secondName}
})

export const getAddresses = (addresses) => ({
  type: GET_ADDRESSES,
  payload: addresses
});

export const fetchAddresses = () => ({
  type: FETCH_ADDRESSES,
})

export const sendRoute = (address1, address2) => ({
  type: SEND_ROUTE,
  payload: {address1, address2}
})

export const saveCoords = (coords) => ({
  type: SAVE_COORDS,
  payload: coords
})