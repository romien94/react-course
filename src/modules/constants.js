export const BASE_URL = 'https://loft-taxi.glitch.me';
export const HEADERS = { "Content-Type": "application/json" };