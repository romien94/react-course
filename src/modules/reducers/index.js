import {combineReducers} from 'redux';
import auth from './auth'
import profile from './profile';
import map from './map';

export default combineReducers({auth,profile, map});