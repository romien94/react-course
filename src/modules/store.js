import { createStore, applyMiddleware, compose } from "redux";
import rootReducer from "./reducers/index";
// import { authMiddleware } from "./middlewares/authMiddleware";
import {authSaga} from '../modules/sagas/authSaga';
import {profileSaga} from '../modules/sagas/profileSaga';
import {registrationSaga} from '../modules/sagas/registrationSaga';
import {addressesSaga} from '../modules/sagas/addressesSaga';
import {routeSaga} from '../modules/sagas/routeSaga';
import createSagaMiddleware from "redux-saga";
import profile from "./reducers/profile";
// import { profileMiddleware } from "./middlewares/profileMiddleware";

const sagaMiddleware = createSagaMiddleware();


// export const store = createStore(
//   rootReducer,
//   compose(applyMiddleware(authMiddleware), applyMiddleware(profileMiddleware))
// );

export const store = createStore(
    rootReducer,
    compose(applyMiddleware(sagaMiddleware))
)

sagaMiddleware.run(authSaga);
sagaMiddleware.run(profileSaga);
sagaMiddleware.run(registrationSaga);
sagaMiddleware.run(addressesSaga);
sagaMiddleware.run(routeSaga);