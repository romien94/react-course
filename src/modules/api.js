import { BASE_URL, HEADERS } from "./constants";

export const serverLogIn = async (email, password) => {
  return fetch(`${BASE_URL}/auth`, {
    method: "POST",
    headers: HEADERS,
    body: JSON.stringify({
      email,
      password,
    }),
  }).then((res) => res.json());
};

export const serverProfile = async (
  cardNumber,
  expiryDate,
  cardName,
  cvc,
  token
) => {
  return fetch(`${BASE_URL}/card`, {
    method: "POST",
    headers: HEADERS,
    body: JSON.stringify({
      cardNumber,
      expiryDate,
      cardName,
      cvc,
      token,
    }),
  }).then((res) => res.json());
};

export const loadServerProfile = async (token) => {
  return fetch(`https://loft-taxi.glitch.me/card?token=${token}`).then(
    (res) => {
      if (res.status === 200) return res.json();
    }
  );
};

export const serverRegister = async (
  email,
  password,
  name,
  surname,
) => {
  return fetch(`${BASE_URL}/register`, {
    method: "POST",
    headers: HEADERS,
    body: JSON.stringify({
      email,
      password,
      name,
      surname,
    }),
  }).then(res => res.json())
};


export const serverAddresses = async (startingPoint, endingPoint) => {
  return fetch(`${BASE_URL}/addressList`)
  .then(res => res.json())
}

// export const serverRoute = async (address1, address2) => {
//   return fetch(`${BASE_URL}/route`, {
//     method: 'GET',
//     headers: HEADERS,
//     body: JSON.stringify({
//       address1,
//       address2
//     })
//   })
//   .then(res => res.json())
// }

export const serverRoute = async (address1, address2) => {
  // return fetch(`${BASE_URL}/auth`, {
  //   method: "POST",
  //   headers: HEADERS,
  //   body: JSON.stringify({
  //     address1,
  //     address2,
  //   }),
  // }).then((res) => res.json());
  return fetch(`${BASE_URL}/route?address1=${address1}&address2=${address2}`)
  .then(res => res.json())
};
