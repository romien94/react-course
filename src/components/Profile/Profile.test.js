import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import Profile from "./Profile";
import {store} from '../../modules/store';
import {Provider} from 'react-redux';
import { BrowserRouter } from "react-router-dom";

const NewProfile = () => (
  <BrowserRouter>
    <Provider store={store}>
      <Profile />
    </Provider>
  </BrowserRouter>
);

describe("Profile works fine", () => {
  const {getByTestId} = render(<NewProfile/>);

  it('Snapshot matches', () => {
    const mockProfile = renderer.create(<NewProfile/>).toJSON();
    expect(mockProfile).toMatchSnapshot();
  })
})