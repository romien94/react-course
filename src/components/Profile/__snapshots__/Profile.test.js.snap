// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile works fine Snapshot matches 1`] = `
<div>
  <div
    className="app-profile"
  >
    <div
      className="app-profile__top"
    >
      <h2
        className="app-profile__title"
      >
        Профиль
      </h2>
      <p
        className="app-profile__text"
      >
        Способ оплаты
      </p>
    </div>
    <form
      className="app-profile__form app-form"
      onSubmit={[Function]}
    >
      <div
        className="app-form__wrapper"
      >
        <div
          className="app-form__cols"
        >
          <div
            className="app-form__col"
          >
            <div
              className="app-form__row"
            >
              <label
                className="MuiFormLabel-root"
              >
                <span
                  className="app-form__fieldname"
                >
                  Номер карты:
                </span>
                <div
                  className="MuiInputBase-root MuiInput-root MuiInput-underline app-form__input"
                  data-testid="number"
                  onClick={[Function]}
                >
                  <input
                    className="MuiInputBase-input MuiInput-input"
                    name="number"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    type="text"
                    value=""
                  />
                </div>
              </label>
            </div>
            <div
              className="app-form__row"
            >
              <label
                className="MuiFormLabel-root"
              >
                <span
                  className="app-form__fieldname"
                >
                  Срок действия:
                </span>
                <div
                  className="MuiInputBase-root MuiInput-root MuiInput-underline app-form__input"
                  data-testid="date"
                  onClick={[Function]}
                >
                  <input
                    className="MuiInputBase-input MuiInput-input"
                    name="date"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    type="text"
                    value=""
                  />
                </div>
              </label>
            </div>
          </div>
          <div
            className="app-form__col"
          >
            <div
              className="app-form__row"
            >
              <label
                className="MuiFormLabel-root"
              >
                <span
                  className="app-form__fieldname"
                >
                  Имя владельца:
                </span>
                <div
                  className="MuiInputBase-root MuiInput-root MuiInput-underline app-form__input"
                  data-testid="name"
                  onClick={[Function]}
                >
                  <input
                    className="MuiInputBase-input MuiInput-input"
                    name="name"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    type="text"
                    value=""
                  />
                </div>
              </label>
            </div>
            <div
              className="app-form__row"
            >
              <label
                className="MuiFormLabel-root"
              >
                <span
                  className="app-form__fieldname"
                >
                  CVC:
                </span>
                <div
                  className="MuiInputBase-root MuiInput-root MuiInput-underline"
                  data-testid="cvc"
                  onClick={[Function]}
                >
                  <input
                    className="MuiInputBase-input MuiInput-input"
                    name="cvc"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    type="password"
                    value=""
                  />
                </div>
              </label>
            </div>
          </div>
        </div>
        <div
          className="app-form__controls"
        >
          <div
            className="app-form__row"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained button app-form__button"
              data-testid="save"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="submit"
            >
              <span
                className="MuiButton-label"
              >
                Сохранить
              </span>
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
